xdsl/irdl/operations.py

```diff
@@ -1235,16 +1235,16 @@ def verify(self, op: Operation):
         for trait in self.traits:
             trait.verify(op)
 
-    def split_properties(self, attr_dict: dict[str, Attribute]) -> dict[str, Attribute]:
+    def split_properties(
+        self, properties: dict[str, Attribute], attr_dict: dict[str, Attribute]
+    ):
         """
         Remove all entries of an attribute dictionary that are defined as properties
         by the operation definition, and return them in a new dictionary.
         """
-        properties: dict[str, Attribute] = {}
         for property_name in self.properties.keys():
-            if property_name in attr_dict:
+            if property_name in attr_dict and property_name not in properties:
                 properties[property_name] = attr_dict.pop(property_name)
```

Comment:
this is sketchy, I would avoid modifying an argument if possible, let's return new properties instead?

Reply:
Would it make sense to separate this change into its own PR? Seems like it's a matter of initializing the result with properties passed in, can probably be done independently of the other changes being reviewed.

Reply:
This function is now only used in one place, so I could inline it if that is preferable, I'm not really sure what there is to separate into a separate PR

Reply:
Yeah inlining also seems good to me.